### PR TITLE
generator should install rspec

### LINKED ISSUE #409: Template installer should run `rails generate rspec:install`
<a href="https://github.com/bess"></img></a> **Issue by [bess](https://github.com/bess)**
_Tuesday Apr 04, 2017 at 22:17 GMT_
_Originally opened as https://github.com/projecthydra-labs/hyrax/issues/708_

----

### Descriptive summary

Re: hydra-head `master` and likely many of its releases.

### Rationale

When installing e.g. Hyrax via the template installer, after you generate a new work type, rspec tests are generated but they won't run because rails_helper.rb doesn't exist.

### Expected behavior

The template should run `rails generate rspec:install` to create all the files necessary to run rspec

### Actual behavior

```
paz:laevigata bess$ bundle exec rake spec
/Users/bess/.rvm/rubies/ruby-2.3.3/bin/ruby -I/Users/bess/.rvm/gems/ruby-2.3.3@hyrax/gems/rspec-core-3.5.4/lib:/Users/bess/.rvm/gems/ruby-2.3.3@hyrax/gems/rspec-support-3.5.0/lib /Users/bess/.rvm/gems/ruby-2.3.3@hyrax/gems/rspec-core-3.5.4/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
/usr/local/projects/hyrax/laevigata/spec/actors/hyrax/actors/etd_actor_spec.rb:3:in `require': cannot load such file -- rails_helper (LoadError)
```

### Steps to reproduce the behavior

1. Install a new app via the template installer
1. Create a new work type
1. Attempt to run test suite via `rake spec` or `rspec spec`

### PR DESCRIPTION
**Pull request by mcritchlow**
Fixes #409 

I placed this in the `ingest_test_framework` method since it seemed most relevant there. Please let me know if it should instead go in its own method.

### CODE REVIEW

**Comment by mjgiarlo** [comment]
@jcoyne You asked that this be done in hydra-head. This what you had in mind?

@mcritchlow++

File: hydra-core/lib/generators/hydra/head_generator.rb [hunk 1]
```diff
@@ -36,6 +36,8 @@ def add_gems
     def inject_test_framework
       return if options[:'skip-rspec']
 
+      generate 'rspec:install'
+
       application("\n" <<
         "    config.generators do |g|\n" <<
```
Comment (jcoyne) [inline hydra-core/lib/generators/hydra/head_generator.rb]: I wonder if `rspec:install` doesn't do this by itself.

Comment (jcoyne) [inline hydra-core/lib/generators/hydra/head_generator.rb]: It doesn't inject these lines, but they also seem to no longer be needed:
```
16:17 $ bin/rails g rspec:install
Running via Spring preloader in process 82543
      create  .rspec
      create  spec
      create  spec/spec_helper.rb
      create  spec/rails_helper.rb

16:18 $ rails g model Bar
Running via Spring preloader in process 82652
      invoke  active_record
      create    db/migrate/20170512211819_create_bars.rb
      create    app/models/bar.rb
      invoke    rspec
      create      spec/models/bar_spec.rb
```

Comment (mcritchlow) [inline hydra-core/lib/generators/hydra/head_generator.rb]: @jcoyne sorry, i'm not following what the requested changes are. or rather, what's missing from invoking the generate command on `rspec:install`.

when you say 'doesn't inject these lines' do you mean in your own testing the generate command didn't actually create the spec folder, the helpers, etc.? It did in mine, so just wanted to clarify.

Comment (jcoyne) [inline hydra-core/lib/generators/hydra/head_generator.rb]: I'm saying remove lines 41-45, they are no longer necessary.

Comment (mcritchlow) [inline hydra-core/lib/generators/hydra/head_generator.rb]: Ah, gotcha. Thanks.

**Pull request merged**